### PR TITLE
Fix(1429): Typing full URI in Endpoint's name generates bad YAML

### LINKED ISSUE #1429: Typing full URI in Endpoint's name generates bad YAML
### Describe the Bug

YAML supports both the syntax of a full URI with parameters and also, a more YAML structured format where parameters are independent from the URI.

A user may try to use Kaoto to force the parameters in the URI by typing them in the component's name field in the configuration pane.

This results on Kaoto generating bad YAML as shown below.

### Steps to Reproduce the Bug or Issue

Try configuring a Timer action as follows (see image):
image

The source code generated will be similar to:
```yaml
- route:
    id: route-1765
    from:
      uri: timer:yaml?repeatCount=1
      parameters:
        r: ""
        re: ""
        rep: ""
        repe: ""
        repea: ""
        repeat: ""
        repeatC: ""
        repeatCo: ""
        repeatCou: ""
        repeatCoun: ""
        repeatCount: ""
        repeatCount-: ""
      steps:
        - setBody:
            simple: Hello Camel from ${routeId}
        - log: ${body}
```

### Screenshots or Videos

_No response_

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [e.g. 0.30.0]

### PR DESCRIPTION
**Pull request by shivamG640**
Fixes #1429

### CODE REVIEW

File: packages/ui/src/utils/camel-uri-helper.ts [hunk 1]
```diff
@@ -109,7 +109,7 @@ export class CamelUriHelper {
       queryString?.split('&').reduce((parameters, parameter) => {
         if (parameter) {
           const [key, stringValue] = parameter.split('=');
-          parameters[key] = getParsedValue(stringValue);
+          if (isDefined(stringValue) && stringValue !== '') parameters[key] = getParsedValue(stringValue);
```
Comment (lordrip) [inline packages/ui/src/utils/camel-uri-helper.ts:112]:
```suggestion
          if (isDefined(stringValue) && stringValue !== '')  {
            parameters[key] = getParsedValue(stringValue);
          }
```

File: packages/ui/src/models/visualization/flows/support/camel-component-schema.service.ts [hunk 2]
```diff
@@ -464,7 +464,17 @@ export class CamelComponentSchemaService {
     const [pathUri, queryUri] = definition.uri?.split('?') ?? [undefined, undefined];
     if (queryUri) {
       definition.uri = pathUri;
-      Object.assign(definition.parameters, CamelUriHelper.getParametersFromQueryString(queryUri));
+
+      const validParametersFromQueryString =
+        Object.entries(CamelUriHelper.getParametersFromQueryString(queryUri)).reduce((parameters, parameter) => {
+          if (parameter) {
+            const [key, stringValue] = parameter;
+            if (catalogLookup.definition?.properties[key]) parameters[key] = stringValue;
```
Comment (lordrip) [inline packages/ui/src/models/visualization/flows/support/camel-component-schema.service.ts:472]:
```suggestion
            if (catalogLookup.definition?.properties[key]) {
              parameters[key] = stringValue;
            }
```

File: packages/ui/src/models/visualization/flows/support/camel-component-schema.service.ts [hunk 3]
```diff
@@ -464,7 +464,17 @@ export class CamelComponentSchemaService {
...
+          }
+
+          return parameters;
+        }, {} as ParsedParameters) ?? {};
```
Comment (lordrip) [inline packages/ui/src/models/visualization/flows/support/camel-component-schema.service.ts:476]:
```suggestion
        }, {} as ParsedParameters);
```

**Comment by shivamG640** [comment]
Created new PR : https://github.com/KaotoIO/kaoto/pull/1493

**Pull request closed without merging**